### PR TITLE
test: remove coverage ignores and improve coverage to 95%+ across all packages

### PR DESCRIPTION
**Pull request by ifiokjr**
## Summary

Removes all `coverage:ignore-*` directives and improves test coverage to 95%+ across every package in the monorepo.

### Coverage improvements

| Package | Before | After |
|---------|--------|-------|
| solana_kit_rpc_parsed_types | 74.84% | **96.57%** |
| solana_kit_codecs_data_structures | 84.51% | **97.70%** |
| solana_kit_rpc_types | 85.95% | **100%** |
| solana_kit_associated_token_account | 86.41% | **95.65%** |
| solana_kit_compute_budget | 89.08% | **98.74%** |
| solana_kit_transactions | 89.35% | **98.39%** |
| solana_kit_signers | 90.00% | **96.88%** |
| solana_kit_transaction_confirmation | 90.11% | **95.19%** |
| solana_kit_subscribable | 90.20% | **100%** |
| solana_kit_transaction_messages | 91.61% | **95.62%** |
| solana_kit_rpc_spec | 92.16% | **100%** |
| solana_kit_errors | 92.76% | **96.83%** |
| solana_kit_instruction_plans | 92.88% | **95.01%** |
| solana_kit_rpc_transformers | 94.05% | **100%** |
| solana_kit_rpc_api | 94.12% | **99.36%** |
| solana_kit_rpc | 94.16% | **100%** |
| solana_kit_codecs_core | 94.28% | **97.83%** |
| solana_kit_fast_stable_stringify | 97.30% | **100%** |
| solana_kit_codecs_strings | 95.71% | **99.52%** |

### Key changes

- **Removed dead code** in `fast_stable_stringify.dart` (unreachable jsonEncode fallback)
- **Updated coverage thresholds** to 95% minimum for all 48 packages in `config/coverage-risk-tier-thresholds.json`
- **Added 500+ tests** across 20+ packages covering equality/hashCode/toString, codec edge cases, error paths, and constructor variants

### 10 packages at 100% coverage
rpc_spec, rpc, rpc_transformers, rpc_types, subscribable, fast_stable_stringify, codecs_numbers, fixed_points, functional, instructions

### CODE REVIEW

File: .changeset/improve-test-coverage-to-95-percent.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+"solana_kit_rpc_transformers": patch
+"solana_kit_rpc_types": patch
+"solana_kit_signers": patch
+"solana_kit_subscribable": patch
+"solana_kit_transaction_confirmation": patch
+"solana_kit_transaction_messages": patch
+"solana_kit_transactions": patch
+---
+
+Improve test coverage to 95%+ across all packages
```
Comment (ifiokjr) [inline .changeset/improve-test-coverage-to-95-percent.md]: Make it a header. 

```suggestion
# Improve test coverage to 95%+ across all packages
```

File: .changeset/improve-test-coverage-to-95-percent.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Added 500+ tests covering equality/hashCode/toString, codec edge cases,
```
Comment (ifiokjr) [inline .changeset/improve-test-coverage-to-95-percent.md]: The next three lines can all be one line.

File: .changeset/improve-test-coverage-to-95-percent.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+"solana_kit_addresses": patch
```
Comment (ifiokjr) [inline .changeset/improve-test-coverage-to-95-percent.md]: Make all of these `test` types


```suggestion
"solana_kit_addresses": test
```
